src/control/lib/hardware/topology.go

```diff
@@ -34,20 +41,130 @@ func (t *Topology) AllDevices() map[string]*Device {
 	return devsByName
 }
 
+func PrintTopology(t *Topology, output io.Writer) error {
```

Comment:
Don't forget comments for the new exported functions and types

Reply:
Yep, was in a hurry to get things shared last night so hadn't gotten around to that yet, will work on it today. Also realized that I need to fix the unit tests to account for the new PCIBus info.